default.nix

```diff
@@ -1,157 +1,92 @@
 {
-  nixpkgsPath ? ./nixpkgs.nix,
+  nixpkgs ? <nixpkgs>,
+  nixpkgsMozilla ? fetchGit {
+    url = https://github.com/mozilla/nixpkgs-mozilla;
+    rev = "50bae918794d3c283aeb335b209efd71e75e3954";
+  },
   system ? builtins.currentSystem,
-  overlays ? [],
-  crossSystem ? (import <nixpkgs/lib>).systems.examples.musl64,
+  crossSystem ? (import nixpkgs {}).lib.systems.examples.musl64
 }:
 let
-  version = "0.3.0";
-
-  # mozilla nixpkgs rust overlay
-  nixpkgs-mozilla = builtins.fetchGit {
-    url = https://github.com/mozilla/nixpkgs-mozilla;
-    ref = "master";
-    rev = "50bae918794d3c283aeb335b209efd71e75e3954";
-  };
-  rustOverlay = import "${nixpkgs-mozilla}/rust-overlay.nix";
-
-  # bootstrap Nixpkgs with the overlays
-  pkgs = import nixpkgsPath {
+  pkgs = import nixpkgs {
     inherit system crossSystem;
-    overlays = overlays ++ [ rustOverlay (import ./overlay) ];
+    overlays =
+      let
+        rustOverlay = import "${nixpkgsMozilla}/rust-overlay.nix";
+        cargo2nixOverlay = import ./overlay;
+      in
+       [ cargo2nixOverlay rustOverlay ];
   };
+  inherit (pkgs) lib buildPackages;
 
-  inherit (pkgs) lib;
-
-  # openssl supply
-  openssl =
-    pkgs:
-      pkgs.buildPackages.symlinkJoin {
-        name = "openssl";
-        paths = with pkgs.openssl; [out dev];
-      };
-
-  # macos frameworks supply, if on darwin
-  macosFrameworks = if pkgs.stdenv.isDarwin
-    then
-      with pkgs.darwin.apple_sdk.frameworks;
-      [Security CoreServices]
-    else [];
-
-  # choice of rustc
-  rustChannel = pkgs.buildPackages.rustChannelOf {
-    channel = "1.37.0";
+  rustChannel = buildPackages.rustChannelOf {
+    channel = "1.38.0";
```

Comment:
I would bump the Rust version in a separate PR, unless there's a specific reason?